controller/deployments_blackbox_test.go

```diff
@@ -736,21 +748,47 @@ func TestWatchEnvironmentEvents(t *testing.T) {
 
 		var buf []byte
 		for _, item := range testItems {
-			// buffer 256 is an arbitrary choice that fits the test items
-			// Manually unmarshal ws frame. Second element contains length
-			// Object is marshaled as JSON and placed in index 2:length+2
-			buf = make([]byte, 256)
+			// buffer 1024 is an arbitrary choice that fits the test items
+			// Manually unmarshal ws frame; Object is marshaled as JSON
+			// For more info on websocket frames see:
+			// https://tools.ietf.org/html/rfc6455#section-5.2
+			buf = make([]byte, 1024)
 			conn.Read(buf)
+			var startPos int
 			frameLength := int(buf[1])
-			startPos := 2
+			if frameLength == 126 {
+				frameLength = int(buf[2])*256 + int(buf[3])
+				startPos = 4
+			} else {
+				startPos = 2
+			}
 			endPos := startPos + frameLength
-			var m testWatchItem
-			err = websocket.JSON.Unmarshal(buf[startPos:endPos], 1, &m)
-			assert.Equal(t, m, item)
+			var actual controller.DeploymentsEvent
+			err = websocket.JSON.Unmarshal(buf[startPos:endPos], 1, &actual)
+
+			expected, _ := transformItem(item)
+			assert.Equal(t, *expected, actual)
 		}
+
+		conn.Close()
 	})
 }
 
+func transformItem(item interface{}) (*controller.DeploymentsEvent, error) {
+	if event, ok := item.(*v1.Event); ok {
```

Comment:
Item is guaranteed to be a `v1.Event` isn't it? We could get rid of the type-assertion and error return value here.